fix: certificate revision list in Django Admin Certificate model

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#2700 

#### What's this PR do?
- Filters only certificate revision pages from CMS for the list in Course Run Certificate model in Django Admin
- Also adds some validation if a user attempts to add the wrong data

#### How should this be manually tested?
- You will need to create some courses along with course pages in CMS and their certificate pages
- Now, On master, Try to open `Course Run Certificates` model in Django Admin
- Open the `Certificate Revision` dropdown, You will see all the pages ever created in CMS
- Now shift to this branch, Notice, You should only see certificate revision pages in this drop-down
- The check, that you see the following errors:
    - Select a course run, Select certificate revision of some other course run (You should see an invalid revision error)
    -  Select a course run that doesn't have any course page in CMS (You should see an error that there is no CMS page for the course)
    - Now, Create a course page for this, But not the certificate page. Now click save. (You should see an error message that the certificate page is not present in the CMS)

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
<img width="1434" alt="image" src="https://github.com/mitodl/mitxpro/assets/34372316/c23d2f94-981f-486f-a070-5a7888e2c64d">


#### What GIF best describes this PR or how it makes you feel?
(Optional)

asadali145 reviewed (approved):
Just a couple of suggestions. Looks good to me.

arslanashraf7 commented:
@asadali145 I've addressed your feedback in https://github.com/mitodl/mitxpro/pull/2701/commits/48f4d5b19e19e5d26a09f7aa11a3a90dd00e50df. Would you like to take anothor look or should I merge?
